core/glrender/src/main/java/edu/cmu/cs/dennisc/render/gl/imp/adapters/GlrBox.java

```diff
@@ -55,266 +55,195 @@
-      this.yMin = owner.yMinimum.getValue();
-      setIsGeometryChanged(true);
-    } else if (property == owner.yMaximum) {
-      this.yMax = owner.yMaximum.getValue();
-      setIsGeometryChanged(true);
-    } else if (property == owner.zMinimum) {
-      this.zMin = owner.zMinimum.getValue();
-      setIsGeometryChanged(true);
-    } else if (property == owner.zMaximum) {
-      this.zMax = owner.zMaximum.getValue();
-      setIsGeometryChanged(true);
-    } else {
-      super.propertyChanged(property);
-    }
-  }
-
-  private double xMin;
-  private double xMax;
-  private double yMin;
-  private double yMax;
-  private double zMin;
-  private double zMax;
+public class GlrBox extends GlrShape<Box> {
+    private void glBox(Context c, boolean isLightingEnabled, boolean isSubElementRequired, boolean isTexturingEnabled) {
+        //todo: revist vertex ordering
+        int id = 0;
+        double[][] leftFace = {{this.xMin, this.yMin, this.zMax}, {this.xMin, this.yMax, this.zMax}, {this.xMin, this.yMax, this.zMin}, {this.xMin, this.yMin, this.zMin}};
+        double[][] rightFace = {{this.xMax, this.yMin, this.zMin}, {this.xMax, this.yMax, this.zMin}, {this.xMax, this.yMax, this.zMax}, {this.xMax, this.yMin, this.zMax}};
+        double[][] bottomFace = {{this.xMin, this.yMin, this.zMin}, {this.xMax, this.yMin, this.zMin}, {this.xMax, this.yMin, this.zMax}, {this.xMin, this.yMin, this.zMax}};
+        double[][] topFace = {{this.xMin, this.yMax, this.zMax}, {this.xMax, this.yMax, this.zMax}, {this.xMax, this.yMax, this.zMin}, {this.xMin, this.yMax, this.zMin}};
+        double[][] frontFace = {{this.xMin, this.yMax, this.zMin}, {this.xMax, this.yMax, this.zMin}, {this.xMax, this.yMin, this.zMin}, {this.xMin, this.yMin, this.zMin}};
+        double[][] backFace = {{this.xMin, this.yMin, this.zMax}, {this.xMax, this.yMin, this.zMax}, {this.xMax, this.yMax, this.zMax}, {this.xMin, this.yMax, this.zMax}};
+        normal3dMap.put("left", leftNormal3d);
+        normal3dMap.put("right", rightNormal3d);
+        normal3dMap.put("bottom", bottomNormal3d);
+        normal3dMap.put("top", topNormal3d);
+        normal3dMap.put("front", frontNormal3d);
+        normal3dMap.put("back", backNormal3d);
+        texCoordMap.put("left", leftTexCoord);
+        texCoordMap.put("right", rightTexCoord);
+        texCoordMap.put("bottom", bottomTexCoord);
+        texCoordMap.put("top", topTexCoord);
+        texCoordMap.put("front", frontTexCoord);
+        texCoordMap.put("back", backTexCoord);
+        vertexMap.put("left", leftFace);
+        vertexMap.put("right", rightFace);
+        vertexMap.put("bottom", bottomFace);
+        vertexMap.put("top", topFace);
+        vertexMap.put("front", frontFace);
+        vertexMap.put("back", backFace);
+
+        c.gl.glBegin(GL_QUADS);
+
+        // xMin face
+        // c.gl.glColor3d( 1,1,1 );
+        if (isSubElementRequired) {
+            c.gl.glLoadName(id++);
```

Comment:
In my previous comment about incrementing this I had intended it to become part of the method call:
`drawBox(c, id++…`
But this works.